test/swe2d/test_rossby_wave.py

```diff
@@ -0,0 +1,400 @@
+    stepper = options.get('timestepper_type')
+
+    # Evaluate error metrics
+    metrics = compute_error_metrics(ref_list, reference_refinement_level, **options)
+
+    # Save metrics to .json file for model comparison
+    if options.get('model_comparison'):
+        di = create_directory(os.path.join(os.path.dirname(__file__), 'data'))
+        with open(os.path.join(di, 'Thetis_{:s}_{:s}.json'.format(family, stepper)), 'w+') as f:
+            json.dump(metrics, f, ensure_ascii=False)
+        # TODO: Plot convergence of error metrics
+
+    # Check convergence of relative mean peak height and phase speed
+    rtol = 0.01
+    for m in ('h+', 'h-', 'c+', 'c-', 'rms'):
+        for i in range(1, len(ref_list)):
+            slope = metrics[m][i-1]/metrics[m][i] if m == 'rms' else metrics[m][i]/metrics[m][i-1]
+            msg = "{:s}: Divergence of error metric {:s}, expected {:.4f} > 1"
+            assert slope > 1.0 - rtol, msg.format(setup_name, m, slope)
+            print_output("{:s}: error metric {:s} index {:d} PASSED".format(setup_name, m, i))
+
+    # Check magnitude of RMS errors
+    if rms_list is not None:
+        assert len(ref_list) == len(rms_list)
+        for i in range(len(ref_list)):
+            msg = "{:s}: RMS error {:.4e} does not match recorded value, expected {:.4e}"
+            calc, rec = metrics['rms'][i], rms_list[i]
+            assert np.allclose(calc, rec, rtol=rtol), msg.format(setup_name, calc, rec)
+            print_output("{:s}: rms magnitude index {:d} PASSED".format(setup_name, i))
+
+
+def generate_table(family, stepper):
+    head = "|Model  |    dx    |    dt    |    h+    |    h-    |    c+    |    c-    |     rms    |"
+    rule = "|-------|----------|----------|----------|----------|----------|----------|------------|"
+    out = '\n'.join([head, rule])
+    msg = "|{:6s} |{:9.3f} |{:9.3f} |{:9.3f} |{:9.3f} |{:9.3f} |{:9.3f} | {:10.4e} |"
+    msg_roms = "|{:6s} |{:9.3f} |{:9.3f} |{:9.3f} |{:9.3f} |{:9.3f} |{:9.3f} |            |"
+    for model in ('FVCOM', 'ROMS', 'Thetis'):
+        fname = os.path.join('data', model)
+        if model == 'Thetis':
+            fname = '_'.join([fname, family, stepper])
+        with open(fname+'.json', 'r') as f:
+            data = json.load(f)
+            for i in range(len(data['dx'])):
+                vals = (model, data['dx'][i], data['dt'][i],)
+                vals += (data['h+'][i], data['h-'][i], data['c+'][i], data['c-'][i],)
+                m = msg
+                if model == 'ROMS':
+                    m = msg_roms
+                else:
+                    vals += (data['rms'][i],)
+                out = '\n'.join([out, m.format(*vals)])
+    return out+'\n'
+
+
+# ---------------------------
+# standard tests for pytest
+# ---------------------------
+
+@pytest.mark.parametrize('stepper,family,rms',
+                         [
+                             ('CrankNicolson', 'dg-dg', [1.1522e-02, 4.3892e-03]),
```

Comment:
Not sure if it makes sense to test that the rmse is allclose to these values. I'd imagine those smaller digits will wiggle as firedrake/petsc etc is updated. So far we've only tested error < target + tolerance. We could change the strategy to indeed test that the errors do not change, but this could trigger a lot of test failures in the future. Don't know if @stephankramer has any thoughts on this.

Reply:
I agree - small variations are to be expected due to solver tolerances, parallel, and in particular with unstructured meshes (not here) mesh generation. In my experience having tests fail if the error changes at all, just leads to a lot of update work which is then blindly done thus defeating the purpose. In this case the errors are all pretty much the same, so I would just check for an error< 1.3 for all - arguably dg-cg is a _little_ more accurate.